File: frontend/src/app/model/user/role-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var RoleType;
(function (RoleType) {
    /**
     * Allowed to create applications.
     */
    RoleType[RoleType["ROLE_CREATE_APPLICATION"] = 0] = "ROLE_CREATE_APPLICATION";
    /**
     * Allowed to process applications.
     */
    RoleType[RoleType["ROLE_PROCESS_APPLICATION"] = 1] = "ROLE_PROCESS_APPLICATION";
    /**
     * Allowed to make decisions.
     */
    RoleType[RoleType["ROLE_DECISION"] = 2] = "ROLE_DECISION";
    /**
     * Allowed to supervise.
     */
    RoleType[RoleType["ROLE_SUPERVISE"] = 3] = "ROLE_SUPERVISE";
    /**
     * Allowed to check invoices.
     */
    RoleType[RoleType["ROLE_INVOICING"] = 4] = "ROLE_INVOICING";
    /**
     * Allowed to login and view applications
     */
    RoleType[RoleType["ROLE_VIEW"] = 5] = "ROLE_VIEW";
    /**
     * Allowed to change system configuration.
     */
    RoleType[RoleType["ROLE_ADMIN"] = 6] = "ROLE_ADMIN";
})(RoleType = exports.RoleType || (exports.RoleType = {}));

File: backend/common-domain/src/main/java/fi/hel/allu/common/domain/types/RoleType.java
package fi.hel.allu.common.domain.types;

public enum RoleType {
  /**
   * Allowed to create applications.
   */
  ROLE_CREATE_APPLICATION,
  /**
   * Allowed to process applications.
   */
  ROLE_PROCESS_APPLICATION,
  /**
   * Allowed to make decisions.
   */
  ROLE_DECISION,
  /**
   * Allowed to supervise.
   */
  ROLE_SUPERVISE,
  /**
   * Allowed to check invoices.
   */
  ROLE_INVOICING,
  /**
   * Allowed to comment applications and projects and to add attachments.
   */
  ROLE_DECLARANT,
  /**
   * Allowed to login and view applications.
   */
  ROLE_VIEW,
  /**
   * Allowed to change system configuration.
   */
  ROLE_ADMIN;
}
